src/sscan.c

```diff
@@ -1,6 +1,6 @@
-#line 2 "sscan.c"
+#line 2 "src/sscan.c"
```

Comment:
This and the other changes which reference "src/somefile" don't look right to me. I suspect you autogenerated these files when you were in the directory above the "src" directory. Was that intentional?

Reply:
Ping @vitcpp

Reply:
Well, if it's intentional and expected, then I'm fine with it!

Reply:
Just some more explanation.

There are two files **sparse.y** and **sscan.l** that are used to generate (**src/sparse.h**, **src/sparse.c**) and **sscan.c** accordingly. You may remove these generated sources and run make - these files will be generated again. These files are not required to be in the repository. I think, they were added to simplify the build process - lex/yacc tools are not required.

The following commands are used by make to generate these sources:
`
/usr/bin/flex -Psphere -osrc/sscan.c src/sscan.l
yacc -d -p sphere_yy -o src/sparse.c src/sparse.y
`

Some times ago the sources were located in the pgsphere root directory. Then, we moved source files to src subdirectory. It seems, these files were not re-generated after that.  Once we run flex/yacc from the pgsphere root directory, we add src/ prefix to the input file name. As the result, #line directive is changed. It may affect error messages when the compiler finds some build errors after that directive - file and line will be taken from this directive. I do not see any other ways it may affect the functionality somehow.

Some other changes occurred when I used newer versions of lex/yacc. As the result we see some changes in comments, produced by these tools in generated files (different tool versions or urls, some new explanations). These changes should not produce any problems.

There are some new token types in sparse.h file should be investigated:
- YYEMPTY = -2
- YYEOF = 0,                     /* "end of file"  */
- YYerror = 256,                 /* error  */
- YYUNDEF = 257,            /* "invalid token"  */

I'm not sure how they affect the parser functionality. I think, I should investigate it just to make sure everything is ok (tests are passed ok with these changes).

I haven't investigated the changes in sparse.c. I think there is no sense to do it. Because lex/yacc with different versions may produce different code.

My own changes - add TOK_ prefix to some token names (SIGN, INT, FLOAT, EULERAXIS) in input files sparse.y, sscan.l to fix the compilation.

In general, we may remove the generated files and ask user to install lex/yacc to build pgsphere.

Finally, I think to investigate how these new tokens affect the parser to make sure everything is ok.